maven-core/src/main/java/org/apache/maven/plugin/internal/PlexusContainerDefaultDependenciesValidator.java

```diff
@@ -40,14 +41,17 @@ class PlexusContainerDefaultDependenciesValidator extends AbstractMavenPluginDep
         super(pluginValidationManager);
     }
 
-    protected void doValidate(MavenSession mavenSession, MojoDescriptor mojoDescriptor) {
-        boolean pcdPresent = mojoDescriptor.getPluginDescriptor().getDependencies().stream()
-                .filter(d -> "org.codehaus.plexus".equals(d.getGroupId()))
-                .anyMatch(d -> "plexus-container-default".equals(d.getArtifactId()));
+    protected void doValidate(
+            RepositorySystemSession session,
+            Artifact pluginArtifact,
+            ArtifactDescriptorResult artifactDescriptorResult) {
+        boolean pcdPresent = artifactDescriptorResult.getDependencies().stream()
+                .filter(d -> "org.codehaus.plexus".equals(d.getArtifact().getGroupId()))
+                .anyMatch(d -> "plexus-container-default".equals(d.getArtifact().getArtifactId()));
 
         if (pcdPresent) {
             pluginValidationManager.reportPluginValidationIssue(
-                    mavenSession, mojoDescriptor, "Plugin depends on plexus-container-default, which is EOL");
+                    session, pluginArtifact, "Plugin depends on plexus-container-default, which is EOL");
```

Comment:
this looks like it could be a more generic "check for end of life" component checker if there other stuff that is EOLed, too.

Reply:
For that, we'd need some "EOL marker" or some metadata?

Historically, the was p-c-d, that was 14 years ago stopped being developed (essentially)  got drop-in replacement in form of "plexus-shim" (that is sisu plexus). Furthermore, JSR330 migration started around same time, so today, there is really no reason to depend on p-c-d artifact (nor plexus shim, except in test scope, as PlexusTestCase is in there still, but also advisable to move off from that Junir3 class).

Reply:
I meant "we maintain a list of EOLed artifacts here and report them if found in the tree", not "look up one specifically". I guess we can rewrite this if a second "this is EOL" artifact shows up.